controller/array_action/array_mediator_ds8k.py

```diff
@@ -510,9 +509,34 @@ def create_snapshot(self, name, volume_name):
         logger.info("finished creating snapshot '{0}' from volume '{1}'".format(name, volume_name))
         return self._generate_snapshot_response(target_api_volume, volume_name)
 
+    def _delete_flashcopy(self, flascopy_id):
+        try:
+            self.client.delete_flashcopy(flascopy_id)
+        except exceptions.NotFound:
+            raise array_errors.VolumeNotFoundError(flascopy_id)
+        except exceptions.ClientException as ex:
+            logger.error(
+                "Failed to delete flascopy {} on array {}, reason is: {}".format(
+                    flascopy_id,
+                    self.identifier,
+                    ex.details
+                )
+            )
+            raise array_errors.FlashcopyDeletionError(flascopy_id)
+
     def delete_snapshot(self, snapshot_id):
-        # TODO: will need to implement
-        raise NotImplementedError
+        logger.info("Deleting snapshot with id : {0}".format(snapshot_id))
+        volume_id = get_volume_id_from_scsi_identifier(snapshot_id)
+        api_volume = self._get_api_volume_by_id(volume_id)
+        if not api_volume.flashcopy:
+            logger.error(
+                "FlashCopy Mapping not found for target volume: {}".format(api_volume.name))
```

Comment:
would you like us to wait with this review until [remaining create snapshot PR comments](https://github.com/IBM/ibm-block-csi-driver/pull/163/#discussion_r433183461) for this release are resolved?

Reply:
once you feel it's fixed, please comment